test/core/libp2p/protocol/identify_test.cpp

```diff
@@ -0,0 +1,271 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "libp2p/protocol/identify.hpp"
+
+#include <vector>
+
+#include <gtest/gtest.h>
+#include "libp2p/multi/uvarint.hpp"
+#include "libp2p/network/connection_manager.hpp"
+#include "libp2p/network/network.hpp"
+#include "libp2p/protocol/identify/pb/identify.pb.h"
+#include "mock/libp2p/connection/capable_connection_mock.hpp"
+#include "mock/libp2p/connection/stream_mock.hpp"
+#include "mock/libp2p/crypto/key_marshaller_mock.hpp"
+#include "mock/libp2p/host_mock.hpp"
+#include "mock/libp2p/network/network_mock.hpp"
+#include "mock/libp2p/network/router_mock.hpp"
+#include "mock/libp2p/peer/address_repository_mock.hpp"
+#include "mock/libp2p/peer/identity_manager_mock.hpp"
+#include "mock/libp2p/peer/key_repository_mock.hpp"
+#include "mock/libp2p/peer/peer_repository_mock.hpp"
+#include "mock/libp2p/peer/protocol_repository_mock.hpp"
+#include "testutil/literals.hpp"
+
+using namespace libp2p;
+using namespace peer;
+using namespace crypto;
+using namespace protocol;
+using namespace network;
+using namespace connection;
+using namespace kagome::common;
+using namespace multi;
+
+using testing::_;
+using testing::Const;
+using testing::Ref;
+using testing::Return;
+using testing::ReturnRef;
+
+class IdentifyTest : public testing::Test {
+ public:
+  void SetUp() override {
+    // create a Protobuf message, which is to be "read" or written
+    for (const auto &proto : protocols_) {
+      identify_pb_msg_.add_protocols(proto);
+    }
+    identify_pb_msg_.set_observedaddr(
+        std::string{remote_multiaddr_.getStringAddress()});
+    for (const auto &addr : listen_addresses_) {
+      identify_pb_msg_.add_listenaddrs(std::string{addr.getStringAddress()});
+    }
+    identify_pb_msg_.set_publickey(marshalled_pubkey_.data(),
+                                   marshalled_pubkey_.size());
+    identify_pb_msg_.set_protocolversion(kLibp2pVersion);
+    identify_pb_msg_.set_agentversion(kClientVersion);
+
+    pb_msg_len_varint_ = std::make_shared<UVarint>(
+        static_cast<uint64_t>(identify_pb_msg_.ByteSize()));
+    identify_pb_msg_bytes_.insert(
+        identify_pb_msg_bytes_.end(),
+        std::make_move_iterator(pb_msg_len_varint_->toVector().begin()),
+        std::make_move_iterator(pb_msg_len_varint_->toVector().end()));
+    identify_pb_msg_bytes_.insert(identify_pb_msg_bytes_.end(),
+                                  identify_pb_msg_.ByteSize(), 0);
+    identify_pb_msg_.SerializeToArray(
+        identify_pb_msg_bytes_.data() + pb_msg_len_varint_->size(),
+        identify_pb_msg_.ByteSize());
+
+    identify_ =
+        std::make_shared<Identify>(*host_, bus_, id_manager_, key_marshaller_);
+  }
+
+  std::shared_ptr<Host> host_ = std::make_shared<HostMock>();
+  std::shared_ptr<HostMock> host_mock_ =
```

Comment:
I think we can leave only HostMock. 
https://github.com/soramitsu/kagome/pull/164/files#diff-fb9afef29dd323a3ba0c0c8d0479c37fR73
 here you can safely pass `host_mock_`